king_phisher/client/assistants/campaign.py

```diff
@@ -144,6 +220,52 @@ def is_editing_campaign(self):
 	def is_new_campaign(self):
 		return self.campaign_id is None
 
+	@property
+	def _url_thread_is_ready(self):
+		return self._url_thread is None or not self._url_thread.is_alive()
+
+	@property
+	def _server_uses_ssl(self):
+		return any(address['ssl'] for address in self.config['server_config']['server.addresses'])
+
+	def set_kpm_assistant_complete(self):
+		if not self._import_kpm.check():
+			self.assistant.set_page_complete(self.assistant.get_nth_page(self.assistant.get_current_page()), False)
+		else:
+			self.assistant.set_page_complete(self.assistant.get_nth_page(self.assistant.get_current_page()), True)
+
+	def signal_kpm_select_clicked(self, _):
+		dialog = extras.FileChooserDialog('Import Message Configuration', self.parent)
+		dialog.quick_add_filter('King Phisher Message Files', '*.kpm')
+		dialog.quick_add_filter('All Files', '*')
+		response = dialog.run_quick_open()
+		dialog.destroy()
+		if not response:
+			return False
+		self._import_kpm.target_file = response['target_path']
+		self.gobjects['entry_kpm_file'].set_text(self._import_kpm.target_file)
+		self.set_kpm_assistant_complete()
+
+	def signal_kpm_dest_folder_clicked(self, _):
+		dialog = extras.FileChooserDialog('Destination Directory', self.parent)
+		response = dialog.run_quick_select_directory()
+		dialog.destroy()
+		if not response:
+			return False
+		self._import_kpm.dest_folder = response['target_path']
+		self.gobjects['entry_kpm_dest_folder'].set_text(self._import_kpm.dest_folder)
+		self.set_kpm_assistant_complete()
+
+	def signal_kpm_file_clear(self, _):
```

Comment:
Once you get rid of the `KpmImport` class and pull this information at runtime, you should combine this and the `signal_kpm_dest_folder` handlers into a single method like:

```
def signal_kpm_entry_clear(self, entry_widget):
    entry_widget.set_text('')
    self.set_kpm_assistant_complete()
```